src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -353,6 +356,44 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._register(disposable);
 	}
 
+	async showProfileMigrationNotification(): Promise<void> {
+		const platform = this._getPlatformKey();
+		const shouldMigrateToProfile = (!!this._configurationService.getValue(TerminalSettingPrefix.Shell + platform) ||
+			!!this._configurationService.getValue(TerminalSettingPrefix.ShellArgs + platform)) &&
+			!!this._configurationService.getValue(TerminalSettingPrefix.DefaultProfile + platform);
+		if (shouldMigrateToProfile && this._storageService.getBoolean(SHOULD_PROMPT_FOR_PROFILE_MIGRATION_KEY, StorageScope.WORKSPACE, true)) {
+			this._notificationService.prompt(
+				Severity.Info,
+				nls.localize('terminalProfileMigration', "You are using deprecated shell/shellArgs settings, but have a defaultProfile set. Want to create a profile from those and use that instead?"),
```

Comment:
How about this?

This terminal is using the deprecated shell/shellArgs settings, do you want to migrate it to a profile?

[Migrate] [Don't show again]